ui/modal/create_password_modal.go

```diff
@@ -148,6 +157,21 @@ func (cm *CreatePasswordModal) SetDescription(description string) *CreatePasswor
 	return cm
 }
 
+func (cm *CreatePasswordModal) EnableOldPassword(enable bool) *CreatePasswordModal {
+	cm.oldPwdEnabled = enable
+	return cm
+}
+
+func (cm *CreatePasswordModal) OldPasswordHint(hint string) *CreatePasswordModal {
+	cm.oldPasswordEditor.Hint = hint
+	return cm
+}
+
+func (cm *CreatePasswordModal) WithOldPasswordCreated(callback func(oldPassword, password string, m *CreatePasswordModal) bool) *CreatePasswordModal {
+	cm.withOldPasswordcallback = callback
+	return cm
+}
```

Comment:
Why do we have an old password callback function? It is my believe that you want to implement a change password functionality. if this is true, then you dont need this.

Use the password modal to collect the old password, and if it checkout, use the create password modal for the new password.

this was how it was implemented on the settings page for start up wallet.

```
for pg.changeStartupPass.Clicked() {
		modal.NewPasswordModal(pg.Load).
			Title("Confirm current startup password").
			Hint("Current startup password").
			NegativeButton(values.String(values.StrCancel), func() {}).
			PositiveButton(values.String(values.StrConfirm), func(password string, pm *modal.PasswordModal) bool {
				go func() {
					var error string
					err := pg.wal.GetMultiWallet().VerifyStartupPassphrase([]byte(password))
					if err != nil {
						if err.Error() == dcrlibwallet.ErrInvalidPassphrase {
							error = "Invalid password"
						} else {
							error = err.Error()
						}
						pm.SetError(error)
						pm.SetLoading(false)
						return
					}
					pm.Dismiss()

					// change password
					modal.NewCreatePasswordModal(pg.Load).
						Title(values.String(values.StrCreateStartupPassword)).
						EnableName(false).
						PasswordHint("New startup password").
						ConfirmPasswordHint("Confirm new startup password").
						PasswordCreated(func(walletName, newPassword string, m *modal.CreatePasswordModal) bool {
							go func() {
								err := pg.wal.GetMultiWallet().ChangeStartupPassphrase([]byte(password), []byte(newPassword), dcrlibwallet.PassphraseTypePass)
								if err != nil {
									m.SetError(err.Error())
									m.SetLoading(false)
									return
								}
								pg.Toast.Notify("Startup password changed")
								m.Dismiss()
							}()
							return false
						}).Show()

				}()

				return false
			}).Show()
		break
	}
```

Reply:
Have to confirm new password field is make sure user have correct one new, like almost app have.

Reply:
Two  steps to do that? but ok now.

Reply:
We are planning add some security to the password modal reason why we have one for setting password and another for verifying it.